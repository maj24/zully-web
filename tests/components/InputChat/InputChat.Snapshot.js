import React from 'react';
import {shallow} from 'enzyme';
import InputChat from '../../../src/shared/components/form/InputChat';
import UploadImageChatButton from '../../../src/dashboard/project/UploadImageChatButton';
import ImageUpload from '../../../src/dashboard/project/ImageUpload';
import {Input} from 'antd';

const setup = propsOverrides => {
  const props = Object.assign({
    placeholder: '',
    onSendMessageRequest: jest.fn(),
    onSendImageRequest: jest.fn(),
  }, propsOverrides);

  const inputChat = shallow(<InputChat {...props}/>);

  return {
    inputChat,
  };
};


describe('InputChat Component', () => {

  test('renders', () => {
    const { inputChat } = setup();
    expect(inputChat).toMatchSnapshot();
  });
});


