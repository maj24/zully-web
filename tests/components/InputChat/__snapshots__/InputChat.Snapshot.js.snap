// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputChat Component renders 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="inner-input"
>
    <ImageUploadContainer
        enabled={true}
        formatTypes={
            Array [
                "*",
              ]
        }
        imgShape="rectangular"
        maxSize={20}
        multifile={false}
        onChange={[Function]}
        showUploadList={false}
    >
        <UploadImageChatButton />
    </ImageUploadContainer>
    <Input
        autosize={false}
        disabled={false}
        onChange={[Function]}
        onPressEnter={[Function]}
        placeholder=""
        prefixCls="ant-input"
        type="text"
        value=""
    />
</div>,
  "nodes": Array [
    <div
      className="inner-input"
>
      <ImageUploadContainer
            enabled={true}
            formatTypes={
                  Array [
                        "*",
                      ]
            }
            imgShape="rectangular"
            maxSize={20}
            multifile={false}
            onChange={[Function]}
            showUploadList={false}
      >
            <UploadImageChatButton />
      </ImageUploadContainer>
      <Input
            autosize={false}
            disabled={false}
            onChange={[Function]}
            onPressEnter={[Function]}
            placeholder=""
            prefixCls="ant-input"
            type="text"
            value=""
      />
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <InputChatForm
        onSendImageRequest={[Function]}
        onSendMessageRequest={[Function]}
        placeholder=""
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": InputChatForm {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "handleInputChange": [Function],
        "handleOnPressEnter": [Function],
        "onImageStatusChange": [Function],
        "props": Object {
          "onSendImageRequest": [Function],
          "onSendMessageRequest": [Function],
          "placeholder": "",
        },
        "refs": Object {},
        "state": Object {
          "text": "",
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="inner-input"
>
          <ImageUploadContainer
                    enabled={true}
                    formatTypes={
                              Array [
                                        "*",
                                      ]
                    }
                    imgShape="rectangular"
                    maxSize={20}
                    multifile={false}
                    onChange={[Function]}
                    showUploadList={false}
          >
                    <UploadImageChatButton />
          </ImageUploadContainer>
          <Input
                    autosize={false}
                    disabled={false}
                    onChange={[Function]}
                    onPressEnter={[Function]}
                    placeholder=""
                    prefixCls="ant-input"
                    type="text"
                    value=""
          />
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="inner-input"
>
          <ImageUploadContainer
                    enabled={true}
                    formatTypes={
                              Array [
                                        "*",
                                      ]
                    }
                    imgShape="rectangular"
                    maxSize={20}
                    multifile={false}
                    onChange={[Function]}
                    showUploadList={false}
          >
                    <UploadImageChatButton />
          </ImageUploadContainer>
          <Input
                    autosize={false}
                    disabled={false}
                    onChange={[Function]}
                    onPressEnter={[Function]}
                    placeholder=""
                    prefixCls="ant-input"
                    type="text"
                    value=""
          />
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <InputChatForm
    onSendImageRequest={[Function]}
    onSendMessageRequest={[Function]}
    placeholder=""
/>,
}
`;
